ssg/oval.py

```diff
@@ -197,38 +196,49 @@ def main():
 
     # append each major element type, if it has subelements
     for element in [definitions, tests, objects, states, variables]:
-        if list(element) > 0:
+        if list(element):
             ovaltree.append(element)
+
     # re-map all the element ids from meaningful names to meaningless
     # numbers
     testtranslator = IDTranslator("scap-security-guide.testing")
     ovaltree = testtranslator.translate(ovaltree)
     (ovalfile, fname) = tempfile.mkstemp(prefix=defname, suffix=".xml")
     os.write(ovalfile, ET.tostring(ovaltree))
     os.close(ovalfile)
+
+    cmd = "oscap oval eval --results " + fname + "-results " + fname
```

Comment:
Shouldn't it be a list instead? See https://docs.python.org/3/library/subprocess.html#subprocess.Popen

Reply:
I think if it was a list, we could get rid of 'shell=True' below, which has "Security considerations".  https://docs.python.org/3/library/subprocess.html#security-considerations

Reply:
Sure, I'll add this in this PR. It is a `str` because that's what it was [currently in master](https://github.com/OpenSCAP/scap-security-guide/blob/master/ssg/oval.py#L213) (I moved it up to add the non-error `cmd` statement). I think I'll also simplify the `print(..)` on new-line number 228.